parser/types/field_type.go

```diff
@@ -115,10 +121,40 @@ func (ft *FieldType) SetFlen(flen int) {
 	ft.flen = flen
 }
 
+func (ft *FieldType) SetFlenUnderLimit(flen int) {
+	if ft.tp == mysql.TypeNewDecimal {
+		ft.flen = mathutil.Min(flen, mysql.MaxDecimalWidth)
+	} else {
+		ft.flen = flen
+	}
+}
+
 func (ft *FieldType) SetDecimal(decimal int) {
 	ft.decimal = decimal
 }
 
+func (ft *FieldType) SetDecimalUnderLimit(decimal int) {
+	if ft.tp == mysql.TypeNewDecimal {
+		ft.decimal = mathutil.Min(decimal, mysql.MaxDecimalScale)
+	} else {
+		ft.decimal = decimal
+	}
+}
+
+func (ft *FieldType) UpdateFlenAndDecimalUnderLimit(old *FieldType, deltaDecimal int, deltaFlen int) {
```

Comment:
I think you should check the ft's type first? And only apply these limit when the type is decimal

Reply:
ok, check the decimal type first, if not, directly return.